MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/MotionPhotoReader.java

```diff
@@ -0,0 +1,296 @@
+        fileInputStream = new FileInputStream(file);
+        FileDescriptor fd = fileInputStream.getFD();
+        int videoOffset = motionPhotoInfo.getVideoOffset();
+        lowResExtractor.setDataSource(fd, file.length() - videoOffset, videoOffset);
+
+        // Find the video track and create an appropriate media decoder
+        for (int i = 0; i < lowResExtractor.getTrackCount(); i++) {
+            MediaFormat format = lowResExtractor.getTrackFormat(i);
+            String mime = format.getString(MediaFormat.KEY_MIME);
+            assert mime != null;
+            if (mime.startsWith("video/")) {
+                lowResExtractor.selectTrack(i);
+                videoFormat = format;
+                lowResDecoder = MediaCodec.createDecoderByType(mime);
+                break;
+            }
+        }
+
+        // Make sure the Android version is capable of supporting MediaCodec callbacks
+        if (Build.VERSION.SDK_INT < M) {
+            Log.e("MotionPhotoReader", "Insufficient Android build version");
+            return;
+        }
+
+        lowResDecoder.setCallback(new MediaCodec.Callback() {
+
+            @Override
+            public void onInputBufferAvailable(@NonNull MediaCodec codec, int index) {
+                boolean result = availableInputBuffers.offer(index);
+                Log.d(TAG, "Input buffers: " + availableInputBuffers.toString());
+            }
+
+            @Override
+            public void onOutputBufferAvailable(@NonNull MediaCodec codec,
+                                                int index,
+                                                @NonNull MediaCodec.BufferInfo info) {
+                Bundle bufferData = new Bundle();
+                bufferData.putInt("BUFFER_INDEX", index);
+                bufferData.putLong("TIMESTAMP_US", info.presentationTimeUs);
+                boolean result = availableOutputBuffers.offer(bufferData);
+                Log.d(TAG, "Output buffers: " + availableOutputBuffers.toString());
+            }
+
+            @Override
+            public void onError(@NonNull MediaCodec codec, @NonNull MediaCodec.CodecException e) {
+                Log.e(TAG, "Error while decoding", e);
+            }
+
+            @Override
+            public void onOutputFormatChanged(@NonNull MediaCodec codec,
+                                              @NonNull MediaFormat format) {
+
+            }
+        }, mediaHandler);
+
+        lowResDecoder.configure(videoFormat, surface, null, 0);
+        lowResDecoder.start();
+    }
+
+    /**
+     * Sets up and starts a new handler thread for managing frame advancing calls and available buffers.
+     */
+    private void startBufferThread() {
+        bufferProcessor = new BufferProcessor(
+                lowResExtractor,
+                lowResDecoder,
+                availableInputBuffers,
+                availableOutputBuffers
+        );
+    }
```

Comment:
this is very short. can we replace all `startBufferThread()` calls with the `bufferProcessor = ...` assignment and then delete `startBufferThread`?